packages/block-editor/src/store/selectors.js

```diff
@@ -2039,13 +2039,52 @@ export const getInserterItems = createRegistrySelector( ( select ) =>
 
 			const blockTypeInserterItems = getBlockTypes()
 				.filter( ( blockType ) =>
-					canIncludeBlockTypeInInserter(
-						state,
-						blockType,
-						rootClientId
-					)
+					hasBlockSupport( blockType, 'inserter', true )
 				)
-				.map( buildBlockTypeInserterItem );
+				.map( buildBlockTypeInserterItem )
+				.reduce( ( accumulator, item ) => {
+					item.rootClientId = rootClientId;
+
+					while (
+						! canInsertBlockTypeUnmemoized(
+							state,
+							item.name,
+							item.rootClientId
+						)
+					) {
+						if ( ! item.rootClientId ) {
+							const { sectionRootClientId } = unlock(
+								getSettings( state )
+							);
+							if (
+								sectionRootClientId &&
+								canInsertBlockTypeUnmemoized(
+									state,
+									item.name,
+									sectionRootClientId
+								)
+							) {
+								item.rootClientId = sectionRootClientId;
+							} else {
+								delete item.rootClientId;
+							}
+							break;
+						} else {
+							const parentClientId = getBlockRootClientId(
+								state,
+								item.rootClientId
+							);
+							item.rootClientId = parentClientId;
+						}
+					}
+
+					// We could also add non insertable items and gray them out.
```

Comment:
That would add no value.

Reply:
It does if you have blocks installed or created and you don't know why they're not there. But we can talk about this after.